From a1e06831361717d88279d1b06f43640e2795ee4f Mon Sep 17 00:00:00 2001
From: "Nahulanthran, Sanjeev" <sanjeev.nahulanthran@intel.com>
Date: Fri, 14 Sep 2018 11:28:58 +0800
Subject: [PATCH 13/98] Addition of ACPI IDs for Machine Driver loading

Adding ACPI IDs for BroadMarket and IVI machine driver dynamic
loading. Additions were made to soc-acpi-intel-bxt-match.c and
covers the following codecs:
	1. wm8731      - INT345A
	2. aic3107M    - INT345B
	3. aic3107S    - INT345C
	4. ivi_hu      - INT344A
	5. ivi_generic - INT344B
	6. ivi_rse     - INT343A

Change-Id: Icb3baeac3b62fc1c597f437008df459376105a61
Signed-off-by: Nahulanthran, Sanjeev <sanjeev.nahulanthran@intel.com>
---
 sound/soc/intel/common/soc-acpi-intel-bxt-match.c | 30 +++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/sound/soc/intel/common/soc-acpi-intel-bxt-match.c b/sound/soc/intel/common/soc-acpi-intel-bxt-match.c
index f39386e5..421fd39 100644
--- a/sound/soc/intel/common/soc-acpi-intel-bxt-match.c
+++ b/sound/soc/intel/common/soc-acpi-intel-bxt-match.c
@@ -17,8 +17,13 @@
 struct snd_soc_acpi_mach snd_soc_acpi_intel_bxt_machines[] = {
 	{
 		.id = "INT343A",
+#if IS_ENABLED(CONFIG_SND_SOC_INTEL_BXTP_IVI_RSE_MACH)
+		.drv_name = "bxt_ivi_rse_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+#else
 		.drv_name = "bxt_alc298s_i2s",
 		.fw_filename = "intel/dsp_fw_bxtn.bin",
+#endif
 	},
 	{
 		.id = "DLGS7219",
@@ -51,6 +56,31 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_bxt_machines[] = {
 		.sof_tplg_filename = "intel/sof-apl-tdf8532.tplg",
 		.asoc_plat_name = "0000:00:0e.0",
 	},
+	{
+		.id = "INT344A",
+		.drv_name = "bxt_ivi_hu_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+	},
+	{
+		.id = "INT344B",
+		.drv_name = "bxt_ivi_generic_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+	},
+	{
+		.id = "INT345A",
+		.drv_name = "lhcrb_wm8731_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+	},
+	{
+		.id = "INT345B",
+		.drv_name = "lhcrb_aic3107M_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+	},
+	{
+		.id = "INT345C",
+		.drv_name = "lhcrb_aic3107S_i2s",
+		.fw_filename = "dsp_fw_release.bin",
+	},
 	{},
 };
 EXPORT_SYMBOL_GPL(snd_soc_acpi_intel_bxt_machines);
-- 
1.9.1

